### PR TITLE
GROWTH-143 - Build dataset for supporting conversion event optimization

### PR DESCRIPTION
**Pull request by kwindau**
Checklist for reviewer:

- [ ] Commits should reference a bug or github issue, if relevant (if a bug is referenced, the pull request should include the bug number in the title).
- [ ] If the PR comes from a fork, trigger integration CI tests by running the [Push to upstream workflow](https://github.com/mozilla/bigquery-etl/actions/workflows/push-to-upstream.yml) and provide the `<username>:<branch>` of the fork as parameter. The parameter will also show up
in the logs of the `manual-trigger-required-for-fork` CI task together with more detailed instructions.
- [ ] If adding a new field to a query, ensure that the schema and dependent downstream schemas have been updated.
- [ ] When adding a new derived dataset, ensure that data is not available already (fully or partially) and recommend extending an existing dataset in favor of creating new ones. Data can be available in the [bigquery-etl repository](https://github.com/mozilla/bigquery-etl), [looker-hub](https://github.com/mozilla/looker-hub) or in [looker-spoke-default](https://github.com/mozilla/looker-spoke-default/tree/e1315853507fc1ac6e78d252d53dc8df5f5f322b).

For modifications to schemas in restricted namespaces (see [`CODEOWNERS`](https://github.com/mozilla/bigquery-etl/blob/main/CODEOWNERS)):
- [ ] Follow the [change control procedure](https://docs.google.com/document/d/1TTJi4ht7NuzX6BPG_KTr6omaZg70cEpxe9xlpfnHj9k/edit#heading=h.ttegrcfy18ck)

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/DENG-3912)

### CODE REVIEW

File: sql/moz-fx-data-shared-prod/google_ads_derived/conversion_event_categorization_v1/query.sql [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+  SELECT
+    client_id,
+  -- we should use their actual first date as their cohort date
+    cfs.first_seen_date,
+    cfs.attribution_campaign,
+    cfs.attribution_content,
+    cfs.attribution_dltoken,
+    cfs.attribution_medium,
+    cfs.attribution_source,
+    cls.report_date,
```
Comment (irrationalagent) [inline sql/moz-fx-data-shared-prod/google_ads_derived/conversion_event_categorization_v1/query.sql:78]: Talked about this in the meeting. I think we want to report a client as `FALSE` in the conversion events if they haven't sent a main ping (they have no rows from clients_last_seen, instrumented in the condition below as `cls.first_seen_date IS NULL`. We then write their `report_date` as their first_seen_date from CFS plus 27 days. 

```
COALESCE(cls.report_date, IF((cls.report_date IS NULL) AND (cls.first_seen_date IS NULL) AND (DATE_DIFF(CURRENT_DATE, cfs.first_seen_date, DAY) >= 27), DATE_ADD(cfs.first_seen_date, INTERVAL 27 DAY), NULL) AS report_date
```

File: sql/moz-fx-data-shared-prod/google_ads_derived/conversion_event_categorization_v1/metadata.yaml [hunk 2]
```diff
@@ -0,0 +1,24 @@
+friendly_name: Conversion Event Categorization
+description: |-
+  Classifies conversion events
+owners:
+- kwindau@mozilla.com
+labels:
+  incremental: false
```
Comment (irrationalagent) [inline sql/moz-fx-data-shared-prod/google_ads_derived/conversion_event_categorization_v1/metadata.yaml:7]: in the meeting we talked about making this incremental on `report_date`

**Review by irrationalagent — CHANGES_REQUESTED** [review]
talked about the reporting cadence in the meeting. added a few comments

**Comment by kwindau** [comment]
Closing & opening a new PR since now has conflicts

**Pull request closed without merging**